speechmatics/client.py

```diff
@@ -348,8 +366,12 @@ async def run(self, stream, transcription_config, audio_settings):
             sys.exit(1)
 
         self.websocket = websocket
-        start_recognition_msg = self._start_recognition(audio_settings)
+        try:
+            start_recognition_msg = self._start_recognition(audio_settings)
+        except ForceEndSession:
+            return
```

Comment:
Should we ensure this also awaits websocket.close()?

Separately from that, we should probably change this entire function to use webosckets.connect via a context manager, so that the close is handled in all these cases automatically.

Reply:
I've refactored the whole thing to use a context manager